java/core-arrow/src/main/java/sleeper/arrow/schema/SchemaBackedByArrow.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.arrow.schema;
+
+
+import org.apache.arrow.vector.types.pojo.Field;
+
+import sleeper.core.schema.Schema;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static sleeper.arrow.schema.SchemaConverter.convertArrowSchemaToSleeperSchema;
+import static sleeper.arrow.schema.SchemaConverter.convertSleeperSchemaToArrowSchema;
+
+public class SchemaBackedByArrow {
+    private final org.apache.arrow.vector.types.pojo.Schema arrowSchema;
+    private final List<String> rowKeyFieldNames;
+    private final List<String> sortKeyFieldNames;
+    private final List<String> valueFieldNames;
```

Comment:
These seem to be unused other than for converting to a Sleeper Schema. We want to be able to get the key fields as Sleeper Field objects.